lychee-lib/src/lib.rs

```diff
@@ -20,7 +20,7 @@
 //!
 //! #[tokio::main]
 //! async fn main() -> Result<()> {
-//!   let client = ClientBuilder::default().build()?;
+//!   let client = ClientBuilder::default().client()?;
```

Comment:
Do you think we could add a `ClientBuilder::client()` function, which just calls `ClientBuilder::default().client()?` internally? Would be easier to read and reduce some boilerplate for the user.

Reply:
I can do
`ClientBuilder::client() == ClientBuilder::builder().build().client() == ClientBuilder::default().client()`

But if you want to supply additional configuration, it becomes
`ClientBuilder::builder().foo().bar().build().client()`
which is too different.

Reply:
The conclusion is I cannot...

`ClientBuilder::builder()` returns a `ClientBuilderBuilder<T1, ..., TN>`, where the generic parameters corresponds to each field of `ClientBuilder`.

When you set options, it just store it in the position of one of the generic parameter.
Then you call `ClientBuilderBuilder<T1, ..., TN>::build()`  to get a `ClientBuilder`.

`ClientBuilder::default()` is equivalent to `ClientBuilder::builder()::build()`.
I've already made a `ClientBuilder::client()`, but it takes a finished `ClientBuilder` with all fields set up.

So we have to accept the change in API.

The generated internal builder from `typed-builder` is ugly.
It's a DFA, of initial state `ClientBuilderBuilder<(), .., ()>` to a finished state `ClientBuiulderBuilder<T1, ..., TN>`.
So for example, say we have `ClientBuilder{ foo: u8, bar: u16}`.
It will generate `ClientBuilderBuilder<T1, T2>`.
And something like
```rust
impl ClientBuilderBuilder<T1, T2> {
  fn foo<U: Into<u8>>(self, value: U) -> Self<u8, T2>;
  fn bar<U: Into<u16>>(self, value: U) -> Self<T1, u16>;
  fn build(self) -> ClientBuilder;
}
```

It's possible to hack around to achieve same API as before. But it's painful.

Reply:
No worries. It's an acceptable tradeoff given the benefits. Thanks for the investigation.